src/Datadog.Trace/Util/ArrayHelper.cs

```diff
@@ -7,8 +9,50 @@ public static T[] Empty<T>()
 #if NET45
             return EmptyArray<T>.Value;
 #else
-            return System.Array.Empty<T>();
+            return Array.Empty<T>();
+#endif
+        }
+
+        public static T[] Copy<T>(T[] array)
+        {
+            int length = array.Length;
+
+            return Copy(array, length, length);
+        }
+
+        public static T[] Copy<T>(T[] array, int newSize, int length)
+        {
+            var target = new T[newSize];
+            Copy(array, target, length);
+
+            return target;
+        }
+
+        public static void Copy<T>(T[] source, T[] target, int length)
+        {
+            Copy(source, target, 0, 0, length);
+        }
+
+        public static void Copy<T>(T[] source, T[] target, int sourcOffset, int targetOffset, int length)
+        {
+            if (typeof(T).IsPrimitive)
+            {
+#if NETCOREAPP3_1_OR_GREATER
+                var span = source.AsSpan(sourcOffset, length);
+                span.CopyTo(new Span<T>(target, targetOffset, length));
+#else
+                Buffer.BlockCopy(source, sourcOffset, target, targetOffset, length);
```

Comment:
Is there strong reasoning to use `Buffer.BlockCopy` here instead of `Array.Copy`? I am not that familiar with the two API's, but it seems like `Array.Copy` would be safer to use in all scenarios unless there's a specific reason for the `Buffer.BlockCopy`

Reply:
According to my performance measurements the order is like that: 
1. Spans < Buffer.BlockCopy < Array.Copy (primitives)
2. Array.Copy < Spans (which is very interesting for non primitives)

The Buffer.BlockCopy works only on primitives and should deal directly with the underlying memory instead with elements like Array.Copy.

Because the logic is situated in the "main flow" in case of jaeger exporter and performance is measurable even for small collections then I thought it's good to use that win but encapsulate it to the helper / method.

Reply:
Understood, makes sense to me